src/store/actions/auth.js

```diff
@@ -86,10 +108,10 @@ export const authGo = (email, password, isRegister) => {
 
     const apiKey = process.env.REACT_APP_FIREBASE_API_KEY;
 
-    let url = `https://www.googleapis.com/identitytoolkit/v3/relyingparty/verifyPassword?key=${apiKey}`;
+    let url = `https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=${apiKey}`;
```

Comment:
Seeing as you're reusing the base and most of the path of the url, it makes sense to have this set to a common config somewhere, and the implementation abstracted:

```
// In another file like, idk, 'global-config.js' or something
export const googleIdentityToolkitUrlDetails = {
  base: 'https://identitytoolkit.googleapis.com',
  accountsPath: '/v1/accounts',
}

// Added to 'utilities'
const buildIdentityToolkitUrlWithApiKey = (apiKey) => (action) => `${googleIdentityToolkitUrlDetails.base}${googleIdentityToolkitUrlDetails.accountsPath}:${action}?key=${apiKey}`;

const apiKey = process.env.REACT_APP_FIREBASE_API_KEY;
export const buildIdentityToolkitUrl = buildIdentityToolkitUrlWithApiKey(apiKey);
```

This allows you to use one clearly defined function elsewhere and not have to understand the implementation in the scope where you use it:

```
import { buildIdentityToolkitUrl } from '../some/path';

let url = buildIdentityToolkitUrl('signInWithPassword');

if(isRegister) {
  url = buildIdentityToolkitUrl('signUp');
}
```

And now this makes sense as a ternary, which clearly defines the behaviour without needing to read how the behaviour is actually processed:
```
const url = isRegister ? buildIdentityToolkitUrl('signUp') : buildIdentityToolkitUrl('signInWithPassword');
```

And that right there is code that is easy to understand, as well as straightforward to debug.

Reply:
@shoelessness Cool, I never even thought of that! I just reworked it a bit to make more sense to me. Do you think making a function to get the API key would be good since it's in a few other places in the codebase, or does that seem like overkill?

Reply:
I think making a function for even most mundane of tasks is the right way to go (in most cases, that is). Even if it's just checking `process.env.....` or something, reading that as opposed to reading `getApiKey()`... I mean I think it's a bit easier to understand. Then if you needed to do something different like get the api key from a config file in a certain environment, at that point, you don't even need to worry about getting the code to be in a function, you just change the function. Approaching every line of code like this means that you have easy to read blocks of execution, with the implementation of the execution hidden away somewhere else where you can reason about it on its own. Good boundaries lead to great results.

Reply:
Ok thanks, added!